shared/data/model/src/main/kotlin/com/ivy/data/model/Tag.kt

```diff
@@ -18,4 +20,16 @@ data class Tag(
     val creationTimestamp: Instant,
     override val lastUpdated: Instant,
     override val removed: Boolean
-) : Syncable, Reorderable
\ No newline at end of file
+) : Syncable, Reorderable
+
+@Suppress("DataClassTypedIDs")
+data class TagAssociation(
+    override val id: TagId,
+    val associatedId: AssociationId,
+    override val lastUpdated: Instant,
+    override val removed: Boolean,
+) : Syncable
+
+@JvmInline
+@Immutable
+value class TagId(override val value: UUID) : UniqueId
```

Comment:
+1 that's bad 😬 We should ensure to never use this in the UI layer. The data model should not depend on Compose

Reply:
Btw, we remove tags from the core `Transaction` model because:
- it's confusing when we fetch them and when not
- it's perf. heavy to always fetch them especially when not needed

Tags should be later added in `:shared:ui:core` in some ViewStateMapper because Tags are primarily UI oriented - we show them in some screens and in some not. Wdyt?

Reply:
If its confusing we can always fetch them

This same problem will exist if we add document support later,
Like Should you fetch DocumentId together with Transaction or not,

we need to manually get Tags in every ViewModel where a List of Transactions is displayed,
so duplicate code vs fetch all time

Reply:
Not manually, we can have a separate use-case and just not do it in the TrnsRepo which is now trash because Tags are fetched in some and not in others methods